src/charts/utilization-bar/utilization-bar-chart.directive.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.charts.directive:pfUtilizationBarChart
+ *
+ * @description
+ *   Directive for rendering a utilization bar chart
+ *   There are three possible fill colors for Used Percentage, dependent on whether or not there are thresholds:<br/>
+ *   <ul>
+ *   <li>When no thresholds exist, or if the used percentage has not surpassed any thresholds, the indicator is blue.
+ *   <li>When the used percentage has surpassed the warning threshold, but not the error threshold, the indicator is orange.
+ *   <li>When the used percentage has surpassed the error threshold, the indicator is is red.
+ *   </ul>
+ *
+ * @param {object} chartData the data to be shown in the utilization bar chart<br/>
+ * <ul style='list-style-type: none'>
+ * <li>.used   - number representing the amount used
+ * <li>.total  - number representing the total amount
+ * </ul>
+ *
+ * @param {object=} chart-title The title displayed on the left-hand side of the chart
+ * @param {object=} layout Various alternative layouts the utilization bar chart may have:<br/>
+ * <ul style='list-style-type: none'>
+ * <li>.type - The type of layout to use.  Valid values are 'regular' (default) displays the standard chart layout,
+ * and 'inline' displays a smaller, inline layout.</li>
+ * <li>.titleLabelWidth - Width of the left-hand title label when using 'inline' layout. Example values are "120px", "20%", "10em", etc..</li>
+ * <li>.usedLabelWidth - Width of the right-hand used label when using 'inline' layout. Example values are "120px", "20%", "10em", etc..</li>
+ * </ul>
+ * @param {string=} used-label-format The format of the used label on the right side of the bar chart. Values may be:<br/>
+ * <ul style='list-style-type: none'>
+ * <li>'actual' - (default) displays the standard label of '(n) of (m) (units) Used'.
+ * <li>'percent' - displays a percentage label of '(n)% Used'.</li>
+ * </ul>
+ * @param {object=} units to be displayed on the chart. Examples: "GB", "MHz", "I/Ops", etc...
+ * @param {object=} thresholds warning and error percentage thresholds used to determine the Usage Percentage fill color
```

Comment:
How many of those attributes would be used in a standard case?

Another, cleaner way of grouping these directives instead of having a single directive with all of the attributes on it would be to have a "parent" directive with child directives for particular types of charts.